pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -133,7 +137,12 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
         _dictionaryCreatorMap.put(columnName, dictionaryCreator);
 
         // Create dictionary
-        dictionaryCreator.build();
+        try {
+          dictionaryCreator.build();
+        } catch (IOException e) {
```

Comment:
The PR description says you want to track IndexOutOfBoundsException whereas you are catching IOE and these are thrown from PinotDataBuffer in the build path. You will want to catch Exception in general and rethrow (as currently done).